File: 백준/레벨Gold IV/11657. 타임머신/타임머신.ts
// @ts-ignore
import fs = require('fs');

const read = fs.readFileSync(0, 'utf-8');
const lines = read.split('\n').map((line: string) => line.split(' ').map(Number));

const bellmanFord = () => {
  const [N, M] = lines[0];

  // 간선리스트 : line[1] ~ line[M]

  // 거리배열 초기화
  const INF = 500 * 10000 + 1;
  const dist = Array(N + 1).fill(INF);
  dist[1] = 0;

  // (노드-1)회 반복
  for (let i = 0; i < N - 1; i++) {
    for (let j = 1; j <= M; j++) {
      const [u, v, w] = lines[j];

      if (dist[u] == INF) continue;

      const nxt = dist[u] + w;
      if (dist[v] > nxt) dist[v] = nxt;
    }
  }

  // 음의 사이클 검사
  let isCycle = false;
  for (let j = 1; j <= M; j++) {
    const [u, v, w] = lines[j];

    if (dist[u] == INF) continue;

    if (dist[v] > dist[u] + w) {
      isCycle = true;
      break;
    }
  }

  // 출력 처리
  const result: string[] = [];
  if (isCycle) {
    result.push('-1');
  } else {
    for (let i = 2; i <= N; i++) {
      result.push(dist[i] === INF ? '-1' : dist[i].toString());
    }
  }
  console.log(result.join('\n'));
};

bellmanFord();

File: 백준/레벨Gold IV/11657. 타임머신/타임머신.py
from sys import stdin

input = stdin.readline

def bellman_ford():
    N, M = map(int, input().split())

    # 간선리스트 초기화
    graph = [0] * M
    for i in range(M):
        u, v, w = map(int, input().split())
        graph[i] = u, v, w

    # 거리배열 초기화
    INF = 500 * 10000 + 1
    dist = [INF] * (N + 1)
    dist[1] = 0    

    # (노드-1)회 반복
    for _ in range(N - 1):
        for u, v, w in graph:
            if dist[u] == INF:
                continue
            nxt = dist[u] + w
            if dist[v] > nxt:
                dist[v] = nxt

    # 음의 사이클 검사
    is_cycle = False
    for u, v, w in graph:
        if dist[u] == INF:
            continue
        if dist[v] > dist[u] + w:
            is_cycle = True
            break

    # 출력 처리
    result = []
    if is_cycle:
        result.append("-1")
    else:
        result.extend(
            str(-1 if dist[i] == INF else dist[i])
            for i in range(2, N + 1)
        )
    print("\n".join(result))

bellman_ford()